trin-core/src/portalnet/overlay_service.rs

```diff
@@ -1152,15 +1301,55 @@ impl<TContentKey: OverlayContentKey + Send, TMetric: Metric + Send>
         Ok(closest_nodes)
     }
 
-    fn send_recursive_findnode(&self, _target: &NodeId) {
-        // TODO: Implement Recursive(iterative) FINDNODE. This is a stub.
+    fn send_iterative_findnode(&mut self, _target: &NodeId) {
+        const DISTANCES_TO_REQUEST_PER_PEER: usize = 1;
+        let mut target = QueryInfo {
+            query_type: QueryType::FindNode(*_target),
+            untrusted_enrs: Default::default(),
+            distances_to_request: DISTANCES_TO_REQUEST_PER_PEER,
+        };
+
+        let target_key: kbucket::Key<NodeId> = target.key();
+        let mut known_closest_peers = Vec::new();
+        {
+            let mut kbuckets = self.kbuckets.write();
+            for closest in kbuckets.closest_values(&target_key) {
+                // Add the known ENR's to the untrusted list
+                target.untrusted_enrs.push(closest.value.enr);
+                // Add the key to the list for the query
+                known_closest_peers.push(closest.key);
+            }
+        }
+
+        if known_closest_peers.is_empty() {
+            warn!("No known_closest_peers found. Return empty result without sending query.");
+        } else {
+            let query_config = FindNodeQueryConfig {
+                parallelism: 3,
+                num_results: 25,
```

Comment:
How did we decide on 25? `sigp/discv5` uses the maximum number of nodes per k-bucket, i.e. `k`.